src/PackageDependencyViewExtension/PackageDependencyView.xaml.cs

```diff
@@ -0,0 +1,238 @@
+    /// </summary>
+    public partial class PackageDependencyView : UserControl
+    {
+        private DependencyTable table = new DependencyTable();
+
+        private WorkspaceModel currentWorkspace;
+
+        /// <summary>
+        /// Event handler for workspaceAdded event
+        /// </summary>
+        /// <param name="obj">workspace model</param>
+        internal void OnWorkspaceChanged(IWorkspaceModel obj)
+        {
+            if (obj is WorkspaceModel)
+            {
+                // Unsubscribe
+                if (currentWorkspace != null)
+                {
+                    currentWorkspace.PropertyChanged -= OnWorkspacePropertyChanged;
+                }
+                DependencyRegen(obj as WorkspaceModel);
+                // Update current workspace
+                currentWorkspace = obj as WorkspaceModel;
+                currentWorkspace.PropertyChanged += OnWorkspacePropertyChanged;
+            }
+        }
+
+        /// <summary>
+        /// Event handler for workspaceRemoved event
+        /// </summary>
+        /// <param name="obj">workspace model</param>
+        internal void OnWorkspaceCleared(IWorkspaceModel obj)
+        {
+            if (obj is WorkspaceModel)
+            {
+                // Clear the dependency table.
+                table.Columns.Clear();
+            }
+        }
+
+        private void OnWorkspacePropertyChanged(object sender, PropertyChangedEventArgs args)
+        {
+            if (args.PropertyName == nameof(currentWorkspace.PackageDependencies))
+                DependencyRegen(currentWorkspace);
+        }
+
+        /// <summary>
+        /// Regenerate dependency table
+        /// </summary>
+        /// <param name="ws">workspace model</param>
+        internal void DependencyRegen(WorkspaceModel ws)
+        {
+            // Clear the dependency table.
+            table.Columns.Clear();
+            foreach (var package in ws.PackageDependencies)
+            {
+                bool matchFound = false;
+                // Check if the target package is installed and loaded.
+                foreach (var loadedPackage in ws.LoadedPackageDependencies)
+                {
+                    if (package.Equals(loadedPackage))
+                    {
+                        table.Columns.Add(new Column()
+                        {
+                            ColumnsData = new ObservableCollection<ColumnData>()
+                            {
+                                new ColumnData(package.Name),
+                                new ColumnData(package.Version.ToString(), 100)
+                            }
+                        });
+                        matchFound = true;
+                        continue;
+                    }
+                }
+                // TODO: Not ideal! O(N * M) complexicty, would like LoadedPackageDependencies to be a dictionary or something with constant package name search time
```

Comment:
Good point. I think we could improve these properties/objects a lot. What if we removed `LoadedPackageDependencies` and just added an `IsLoaded` property to the `PackageDependencyInfo` type?

Reply:
I think there are a few reasons that it might make sense to split `PackageDependencyInfo` into two classes: `PackageInfo` and `PackageDependency`

Reply:
@scottmitchell We have to also not only deal with loaded but also higher version which are compatible and other cases, like lower version exist but not sure if they should satisfy.. Any case, there are some aspects to think more here